DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -89,6 +115,22 @@ public IActionResult Index(RegisterViewModel model)
                     centresDataService.GetActiveCentresAlphabetical(),
                     model.Centre
                 );
+
+                if (model.IsCentreSpecific)
+                {
+                    var centreName = centresDataService.GetCentreName(model.Centre.Value);
+                    // if centreId invalid, then clear centre specific details
+                    if (centreName == null)
+                    {
+                        model.Centre = null;
+                        model.IsCentreSpecific = false;
+                    }
+                    else
+                    {
+                        // otherwise continue centre-specific registration process
+                        ViewBag.CentreName = centreName;
+                    }
+                }

```

Comment:
I'm not sure the logic here is quite right either. In the POST action, I don't see how we hit this code.

Either we are getting our set value from the hidden (which can assume has a value so we won't get to this code as ModelState will be valid) or we are in the non-centre specific journey and they really have tried to post without selecting a centre.

If we are concerned about the hidden field being changed, I think it would be better do a different check for this in the centre specific case outside (before) the ModelState.IsValid check, but I don't think that we should be handling this case anyway. If this value has been changed from its set value in the centre specific journey, this suggests to me that the user is maliciously trying to change centre. (not that that would do them much good since they'd just get rejected anyway)

Reply:
I see. The only way the centreName==null conditional is true is if the user maliciously changes the centreId to a non-existing one, or the centre for some reason becomes inactive or invalid. And this can just give a 500 error. So I will remove that check. We do however need to check isCentreSpecific, and set the CentreName if it is, so I will leave that in.

Reply:
I put the check in because we have to GetCentreName anyway so might as well make sure it worked, but I see how that unnecessarily complicates the code.

Reply:
My point was more that even if a user maliciously changes it to another value (whether it exists or not), this doesn't currently invalidate the model, and so none of this code even runs. Happy to talk about it if that would be easier that going back and forth in comments on the pull request

Reply:
Ah yes, I get it.